### PR TITLE
FIX #4782: adding afCa to r_core_anal_all

### LINKED ISSUE #4782: should afCa be added to aa
afCa autorename and declare vars/args

### PR DESCRIPTION
**Pull request by oddcoder**

### CODE REVIEW

**Comment by Maijin** [comment]
Segfaults in travis

File: libr/core/anal.c [hunk 1]
```diff
@@ -2090,6 +2090,7 @@ R_API int r_core_anal_all(RCore *core) {
 		r_list_foreach (list, iter, entry) {
 			ut64 addr = r_bin_get_vaddr (core->bin, entry->paddr, entry->vaddr);
 			r_core_anal_fcn (core, addr, -1, R_ANAL_REF_TYPE_NULL, depth);
+
```
Comment (radare) [inline libr/core/anal.c]: why an empty line here?

File: libr/core/anal.c [hunk 2]
```diff
@@ -2107,6 +2108,8 @@ R_API int r_core_anal_all(RCore *core) {
 	}
 	/* Set fcn type to R_ANAL_FCN_TYPE_SYM for symbols */
 	r_list_foreach (core->anal->fcns, iter, fcni) {
+
```
Comment (radare) [inline libr/core/anal.c]: why that empty line?

File: libr/core/anal.c [hunk 3]
```diff
@@ -2107,6 +2108,8 @@ R_API int r_core_anal_all(RCore *core) {
...
+		fcn_callconv (core, fcni);
```
Comment (radare) [inline libr/core/anal.c]: probably better to move this after the ->break check

File: libr/anal/bb.c [hunk 4]
```diff
@@ -10,6 +10,7 @@
 R_API RAnalBlock *r_anal_bb_new() {
 	RAnalBlock *bb = R_NEW0 (RAnalBlock);
 	if (!bb) return NULL;
+	memset (bb, 0, sizeof (RAnalBlock));
```
Comment (radare) [inline libr/anal/bb.c]: unnecessary, R_NEW0 already memsets

File: libr/anal/bb.c [hunk 5]
```diff
@@ -171,8 +172,10 @@ R_API void r_anal_bb_set_offset(RAnalBlock *bb, int i, ut16 v) {
 	// the offset of the instruction 0 is not stored because always 0
 	if (i > 0) {
 		if (i >= bb->n_op_pos) {
+			int offset = bb->n_op_pos;
 			bb->n_op_pos = i * 2;
 			bb->op_pos = realloc (bb->op_pos, bb->n_op_pos * sizeof (*bb->op_pos));
+			memset (bb->op_pos + offset, 0, (bb->n_op_pos - offset) * sizeof (*bb->op_pos));
```
Comment (radare) [inline libr/anal/bb.c]: no need to memset just one part. the whole buffer should be cleaned up, or filled when `read`

File: libr/core/anal.c [hunk 6]
```diff
@@ -1613,14 +1613,19 @@ static RList *recurse(RCore *core, RAnalBlock *from, RAnalBlock *dest) {
...
 	ut8 *buf = calloc(1,core->anal->opt.bb_max_size);
 	RListIter *tmp = NULL;
 	RAnalBlock *bb = NULL;
 	int i;
-	if(!core || !fcn || !buf){
+	if(!buf){
 		return;
 	}
 	r_list_foreach (fcn->bbs, tmp, bb) {
+		if (bb->size < 0) continue;
```
Comment (radare) [inline libr/core/anal.c]: check must be if < 1

File: libr/core/anal.c [hunk 7]
```diff
@@ -1613,14 +1613,19 @@ static RList *recurse(RCore *core, RAnalBlock *from, RAnalBlock *dest) {
...
+		buf = realloc(buf, bb->size);
```
Comment (radare) [inline libr/core/anal.c]: missing space

**Comment by radare** [comment]
Please check the tests and fix them with a PR in r2r

**Comment by radare** [comment]
Should be green again after this

**Pull request merged**